browser-sign-in/app/src/main/java/com/okta/android/samples/browser_sign_in/BrowserSignInActivity.java

```diff
@@ -44,89 +49,89 @@ protected void onCreate(Bundle savedInstanceState) {
 
         oktaProgressDialog = new OktaProgressDialog(this);
 
-        ((Button) findViewById(R.id.browser_sign_in_btn)).setOnClickListener(v -> signIn());
+        ((Button) findViewById(R.id.browser_sing_in)).setOnClickListener(v -> signIn());
 
         init();
     }
 
+    @Override
+    protected void onResume() {
+        super.onResume();
+        if (mWebAuth.isInProgress()) {
+            oktaProgressDialog.show();
+        }
+    }
+
     private void init() {
-        oktaProgressDialog.show();
-        mOktaAuth = OktaAppAuth.getInstance(this);
 
-        mOktaAuth.init(
-                this,
-                new OktaAppAuth.OktaAuthListener() {
+        mWebAuth = ServiceLocator.provideWebAuthClient(getApplicationContext());
```

Comment:
I'm not agree that we need provide in this sample activity context. We show user that in this case isn't require activity context, he can use application. But if user want to sign-in with web-browser it require activity context. Also, if you use any DI framework, it's easy to init this object using application context rather than activity.

Reply:
The sample is not using any DI framework though. This doesn't really matter. I'm just saying that we should use what we need in samples. Since this sample can use a more restrictive activity context than a Application context.